compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs

```diff
@@ -369,24 +369,26 @@ getIntegrationTests registerTODO scenarioService (packageDbPath, packageFlags) =
 newtype TestCase = TestCase ((String -> IO ()) -> IO Result)
 
 instance IsTest TestCase where
-  run _ (TestCase r) _ =
-    withLog \log -> do
+  run _ (TestCase r) _ = do
+    (res, log) <- withLog \log -> do
```

Comment:
Suggestion: to avoid confusion with the `log` function taken by the argument of `withLog`, call it `testLog`, just like you call it `buildLog` below.